types/mapcmd/clear.go

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package mapcmd
+
+import (
+	"context"
+	"time"
+
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/spf13/cobra"
+
+	hzcerrors "github.com/hazelcast/hazelcast-commandline-client/errors"
+)
+
+const MapClearExample = `  # Clear all entries of given map.
+  hzc map clear -n mapname`
+
+func NewClear(config *hazelcast.Config) *cobra.Command {
+	var mapName string
+	cmd := &cobra.Command{
+		Use:     "clear [--name mapname]",
+		Short:   "Clear entries of the map",
+		Example: MapClearExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// context timeout can be given according to bulk size of operation
+			// we assume that current payload is same for all hazelcast operations
+			ctx, cancel := context.WithTimeout(cmd.Context(), time.Second*3)
```

Comment:
Why is the 3 second timeout necessary? Same comment for other timeouts introduced in this PR.

Reply:
Removed timeout related context and relied on `cmd.Context()`. [7a991c2](https://github.com/hazelcast/hazelcast-commandline-client/pull/67/commits/7a991c20b78270f67c8e763f4727f6ef1e3a3463).